packages/harpguru-core/src/components/menu-of-scales/utils/reduce-for-new-harpstrata-by-target-scale/reduce-for-new-harpstrata-by-target-scale.ts

```diff
@@ -0,0 +1,25 @@
+import { getHarpStrata, HarpStrata } from 'harpstrata'
+import type { DegreeIds } from 'harpparts'
+
+import { compareActiveIds } from '../../../../utils/compare-active-ids'
+
+export const reduceForNewHarpStrataByTargetScale = (
+  activeHarpStrata: HarpStrata,
+  targetScale: ReadonlyArray<DegreeIds>
+): HarpStrata => {
+  const {
+    apparatus: { tuningId, valvingId },
+    pozitionId,
+    harpKeyId,
+    activeDegreeIds,
+  } = activeHarpStrata
+  if (compareActiveIds(activeDegreeIds, targetScale)) return activeHarpStrata
```

Comment:
I don't think these comparison functions should be called `compare..`. Perhaps `isMatchActiveIds` etc would be better.